peps/pep-0792.rst

```diff
@@ -0,0 +1,371 @@
+
+``deprecated``
+~~~~~~~~~~~~~~
+
+Description: The project is considered obsolete, and may have been superseded
+by another project.
+
+Index semantics:
+
+* The index hosting the project **MUST** allow uploads of new distributions to
+  the project.
+* The index **MUST** offer existing distributions of the project for download.
+
+Installer semantics:
+
+* Installers **MAY** produce warnings about a project's deprecation.
+
+Status markers in the index APIs
+--------------------------------
+
+This PEP defines version 1.4 of the index APIs.
+
+HTML index
+~~~~~~~~~~
+
+The following changes are made to the
+:ref:`simple repository API <packaging:simple-repository-api-base>`:
+
+* The index **SHALL** define the ``pypi:repository-version`` as ``1.4``.
+* The index **SHOULD** add an appropriate ``pypi:project-status`` meta tag, with
+  a ``content`` of the project's status marker. The index **MAY** choose to omit
+  the ``pypi:project-status`` meta tag if the project is marked as ``active``.
+
+For example, the following would be a valid HTML index response for
+``sampleproject`` after is has been marked as ``quarantined``:
+
+.. code-block:: html
+    :emphasize-lines: 5
+
+    <!DOCTYPE html>
+    <html>
+      <head>
+        <meta name="pypi:repository-version" content="1.4">
+        <meta name="pypi:project-status" content="quarantined">
+        <title>Links for sampleproject</title>
+      </head>
+      <body>
+        <h1>Links for sampleproject</h1>
+      </body>
+    </html>
+
+Observe that, per the ``quarantined`` semantics above, the index response
+contains no distribution links for the project.
+
+JSON index
+~~~~~~~~~~
+
+The following changes are made to the
+:ref:`JSON simple index <packaging:simple-repository-api-json>`:
+
+* The index **SHALL** define the ``meta.api-version`` as ``1.4``.
+* The index **SHOULD** include a ``project-status`` key in the JSON response,
+  with a value of the project's status marker. The index **MAY** choose to omit
+  the ``project-status`` key if the project is marked as ``active``.
+
+For example, the following would be a valid JSON index response for
+``sampleproject`` after is has been marked as ``quarantined``:
+
+.. code-block:: json
+    :emphasize-lines: 5
+
+    {
+      "meta": {
+        "api-version": "1.4"
+      },
+      "project-status": "quarantined",
+      "alternate-locations": [],
+      "files": [],
+      "name": "sampleproject",
+      "versions": [
+        "1.2.0",
+        "1.3.0",
+        "1.3.1",
+        "2.0.0",
+        "3.0.0",
+        "4.0.0"
+      ]
+    }
+
+Observe that, like with the HTML index, the JSON response contains no
+distribution links for the ``quarantined`` project.
+
+Future Considerations
+=====================
+
+This PEP defines only four project status markers: ``active``, ``archived``,
+``quarantined``, and ``deprecated``.
+
+Future PEPs (or PyPA standards processes) may define additional project
+status markers, as needed.
```

Comment:
Am I right in understanding that further status markers would require a change to the metadata version?

Reply:
I think so, yeah (I don't feel confident asserting that as a manner of policy, but that seems to be the norm, and would make intuitive sense here). Do you think that makes sense to say explicitly?

Reply:
Perhaps, as it means that the light-touch process can't be used. Thinking again, given this demands behaviour from both indices and installers, it would require a PEP to add a new 'status marker'. 

On that note, it might be worth adding a line on the behaviour for encountering an unknown marker. It should never happen given metadata versions, I believe, but an explict 'fail' or 'ignore' might be helpful.

A